sktime/forecasting/var.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+"""Vector Auto Regressor."""
+__all__ = ["VectorAutoRegression"]
+__author__ = ["Taiwo Owoseni"]
+
+from statsmodels.tsa.api import VAR as _VAR
+from sktime.forecasting.base.adapters import _StatsModelsAdapter
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+import numpy as np
+
+
+class VectorAutoRegression(_StatsModelsAdapter):
+    """
+    A VAR model is a generalisation of the univariate autoregressive.
+
+    A model for forecasting a vector of time series[1].
+
+    Parameters
+    ----------
+    dates: np.ndarray, optional (default=None)
+        An array-like object of datetime objects
+    freq: str, optional (default=None)
+        The frequency of the time-series.
+        Pandas offset or ‘B’, ‘D’, ‘W’, ‘M’, ‘A’, or ‘Q’.
+    missing: str, optional (default='none')
+        A string specifying if data is missing
+
+    References
+    ----------
+    [1] Athanasopoulos, G., Poskitt, D. S., & Vahid, F. (2012).
+    Two canonical VARMA forms: Scalar component models vis-à-vis the echelon form.
+    Econometric Reviews, 31(1), 60–83, 2012.
+
+    Example
+    -------
+    >>> from sktime.forecasting.var import VectorAutoRegression as VAR
+    >>> from sktime.forecasting.model_selection import temporal_train_test_split
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> import pandas as pd
+    >>> index = pd.date_range(start="2005", end="2006-12", freq="M")
+    >>> df = pd.DataFrame(np.random.randint(0, 100, size=(23, 2)),
+    ... columns=list("AB"),
+    ... index=pd.PeriodIndex(index))
+    >>> train, test = temporal_train_test_split(df)
+    >>> sktime_model = VAR()
+    >>> fh = ForecastingHorizon([1, 3, 4, 5, 7, 9])
+    >>> sktime_model.fit(train)
+    VectorAutoRegression(dates=None, freq=None, missing='none')
+    >>> y_pred = sktime_model.predict(fh=fh)
+    """
+
+    _fitted_param_names = ("aic", "fpe", "hqic", "bic")
+
+    _tags = {
+        "scitype:y": "multivariate",
+        "y_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "univariate-only": False,
+    }
+
+    def __init__(self, dates=None, freq=None, missing="none"):
+        # Model params
+        self.dates = dates
```

Comment:
Remove `dates` and `freq` as this is something that is handled by the index of the data frame.

Reply:
removed